dlp/inspectWithCustomRegex.js

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//  title: Inspects strings
+//  description: Inspects a string using custom regex pattern.
+//  usage: node inspectWithCustomRegex.js my-project string minLikelihood maxFindings infoTypes customInfoTypes includeQuote
+
+function main(
+  projectId,
+  string,
+  minLikelihood,
+  maxFindings,
+  infoTypes,
+  customInfoTypes,
+  includeQuote
+) {
+  [infoTypes, customInfoTypes] = transformCLI(infoTypes, customInfoTypes);
+  // [START dlp_inspect_custom_regex]
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Instantiates a client
+  const dlp = new DLP.DlpServiceClient();
+
+  // The project ID to run the API call under
+  // const projectId = 'my-project';
+
+  // The string to inspect
+  // const string = 'Patients MRN 444-5-22222';
+
+  // The minimum likelihood required before returning a match
+  // const minLikelihood = DLP.protos.google.privacy.dlp.v2.Likelihood.POSSIBLE;
+
+  // The maximum number of findings to report per request (0 = server maximum)
+  // const maxFindings = 0;
+
+  // The infoTypes of information to match
+  // See https://cloud.google.com/dlp/docs/concepts-infotypes for more information
+  // about supported infoTypes.
+  // const infoTypes = [{ name: 'EMAIL_ADDRESS' }];
+
+  // The customInfoTypes of information to match
+  // const customInfoTypes = [{ infoType: { name: 'DICT_TYPE' }, dictionary: { wordList: { words: ['foo', 'bar', 'baz']}}},
+  //   { infoType: { name: 'REGEX_TYPE' }, regex: {pattern: '\\(\\d{3}\\) \\d{3}-\\d{4}'}}];
+
+  // Whether to include the matching string
+  // const includeQuote = true;
+
+  async function inspectWithCustomRegex() {
+    // Construct item to inspect
+    const item = {
+      byteItem: {
+        type: DLP.protos.google.privacy.dlp.v2.ByteContentItem.BytesType
+          .TEXT_UTF8,
+        data: Buffer.from(string, 'utf-8'),
+      },
+    };
+
+    // Assigns likelihood to each match
+    customInfoTypes = customInfoTypes.map(customInfoType => {
```

Comment:
Update code, now we are constructing the customInfoType in the code only instead of passing that as a param. Defining the likelihood property there as well.